src/index.ts

```diff
@@ -532,25 +552,51 @@ const avocadoForDir = async (cwd: string) => {
 const avocadoForDevOps = (pr: devOps.PullRequestProperties): asyncIt.AsyncIterableEx<err.Error> =>
   asyncIt.iterable<err.Error>(async function*() {
     // collect all errors from the 'targetBranch'
-    await pr.checkout(pr.targetBranch)
-    const targetMap = await avocadoForDir(pr.workingDir)
+    const diffFiles = await pr.diff()
+    const changedSwaggerFilePath = diffFiles.map(item => item.path)
+
+    const swaggerParentDirs = new Set<string>()
+    changedSwaggerFilePath
+      .map(item => path.dirname(path.resolve(pr.workingDir, item)))
+      .filter(item => item !== pr.workingDir)
+      .every(item => swaggerParentDirs.add(item))
+    const readmeDirs = new Set<string>()
+    for (const item of swaggerParentDirs) {
+      console.log(item)
+      const readmeDir = await findTheNearestReadme(pr.workingDir, item)
+      if (readmeDir !== undefined) {
+        readmeDirs.add(readmeDir)
+      } else {
+        yield {
+          level: 'Error',
+          code: 'MISSING_README',
+          message: 'Can not find readme.md in the folder. If no readme.md file, it will block SDK generation.',
+          folderUrl: item,
+        }
+      }
+    }
 
-    // collect all errors from the 'sourceBranch'
-    await pr.checkout(pr.sourceBranch)
-    const sourceMap = await avocadoForDir(pr.workingDir)
+    for (const dir of readmeDirs) {
+      await pr.checkout(pr.targetBranch)
+      const targetMap = await avocadoForDir(path.resolve(pr.workingDir, dir))
 
-    const fileChanges = await pr.diff()
+      // collect all errors from the 'sourceBranch'
+      await pr.checkout(pr.sourceBranch)
+      const sourceMap = await avocadoForDir(path.resolve(pr.workingDir, dir))
```

Comment:
What if the source branch does not have the dir ?

Reply:
it will return empty map and only report new errors.